### DISCUSSION TITLE
[RFC] Filter improvements

### DISCUSSION THREAD

**Discussion by QxxxGit** [stashapp/stash#2364] [Ideas]
I feel as though the filter could use some really big improvements:

# Multiple filters based on the same filter option

For example, if I go to the images tab, filter for "include tag GIF" but I also want to exclude a sub-tag 'riding' while allowing others, I can't do this. It would make more sense to me if the filter accepted an array of this same filter option. Other random examples would be (under performer) gender. You can only select one of Male, Female, Transmale, Transfemale, and nonbinary -- some people may want to select multiple in this.

I don't think we should update these individual options; rather, we should allow plurality of all filter options.

# Dig-deep functionality

I think it would be super valuable if end-users could dig into and filter values of fields that are linked to other tables. There are countless examples of this already (Scenes tab -> "Performer favorited" filter) but these seem to be manually added.

Whenever new tables are added (audio, reading material, whatever else), there will always be requests for more and more options to filter. This takes up too much development time when that time could be spent elsewhere. Automation of this would be more time efficient, and more flexible for end-users.

So, if I wanted to do the odd search for scenes where the performer(s) have a non-null value for piercings or tattoos, or with a certain eye color, it would be pull scenes with that criteria. You can see why people might want this, but manually adding all of these filter options would get super tedious, and probably bloat the source code.

# So what's next?

Something like this would require both a visual and back-end rework to accept these new features. I don't have a concrete idea of what the front-end would look like, so I'd be interested to hear what you folks could cook up. The goal would be something simple and clean, flexible, and usable on both desktop and mobile.

**Comment by WithoutPants** [comment]
Related issue #2122.
